src/vm/hints/hint_utils.rs

```diff
@@ -275,3 +276,62 @@ pub fn assert_le_felt(
         _ => Err(VirtualMachineError::FailedToGetIds),
     }
 }
+
+//Implements hint:
+// %{
+//     from starkware.cairo.common.math_utils import assert_integer
+//     assert_integer(ids.a)
+//     assert 0 <= ids.a % PRIME < range_check_builtin.bound, f'a = {ids.a} is out of range.'
+// %}
+pub fn assert_nn(
+    vm: &mut VirtualMachine,
+    ids: HashMap<String, BigInt>,
+) -> Result<(), VirtualMachineError> {
+    //Check that ids contains the reference id for 'a' variable used by the hint
+    let a_ref = if let Some(a_ref) = ids.get(&String::from("a")) {
+        a_ref
+    } else {
+        return Err(VirtualMachineError::IncorrectIds(
+            vec![String::from("a")],
+            ids.into_keys().collect(),
+        ));
+    };
+    //Check that 'a' reference id corresponds to a value in the reference manager
+    let a_addr =
+        if let Some(a_addr) = get_address_from_reference(a_ref, &vm.references, &vm.run_context) {
+            a_addr
+        } else {
+            return Err(VirtualMachineError::FailedToGetIds);
+        };
+    //Check that the 'a' id is in memory
+    match vm.memory.get(&a_addr) {
+        Ok(Some(maybe_rel_a)) => {
+            //assert_integer(ids.a)
+            let a = if let &MaybeRelocatable::Int(ref a) = maybe_rel_a {
+                a
+            } else {
+                return Err(VirtualMachineError::ExpectedInteger(a_addr.clone()));
+            };
+            for (name, builtin) in &vm.builtin_runners {
+                //Check that range_check_builtin is present
+                if name == &String::from("range_check") {
+                    match builtin.as_any().downcast_ref::<RangeCheckBuiltinRunner>() {
+                        None => return Err(VirtualMachineError::NoRangeCheckBuiltin),
+                        Some(builtin) => {
+                            // assert 0 <= ids.a % PRIME < range_check_builtin.bound
+                            if bigint!(0) <= a.mod_floor(&vm.prime)
+                                && a.mod_floor(&vm.prime) < builtin._bound
+                            {
+                                return Ok(());
+                            } else {
+                                return Err(VirtualMachineError::ValueOutOfRange(a.clone()));
+                            }
+                        }
+                    }
+                }
+            }
+            Err(VirtualMachineError::NoRangeCheckBuiltin)
+        }
+        _ => Err(VirtualMachineError::FailedToGetIds),
+    }
```

Comment:
We can probably reduce nesting here. E.g.:
```
    let maybe_rel_a = if let Ok(Some(maybe_rel_a)) = vm.memory.get(&a_addr) {
        maybe_rel_a
    } else {
        return Err(VirtualMachineError::FailedToGetIds);
    }
```

Then the loop could only contain the linear search and then just use the result after.

Reply:
Thanks for the advice pelito.
I implemented this modification in this [commit](https://github.com/lambdaclass/cleopatra_cairo/pull/243/commits/8b6db26679cb9e96964a79076eed16d80981e496)